### PR TITLE
imp: WriteUpgradeTry 

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

This PR pulls out the `WriteUpgradeTry` logic from the #3628 UpgradeTry handler.

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/upgrade.go [hunk 1]
```diff
@@ -81,15 +81,30 @@ func (k Keeper) ChanUpgradeTry(
...
 }
 
-// WriteUpgradeTryChannel writes a channel which has successfully passed the UpgradeTry step.
+// WriteUpgradeTryChannel writes a channel which has successfully passed the UpgradeTry handshake step.
 // An event is emitted for the handshake step.
 func (k Keeper) WriteUpgradeTryChannel(
 	ctx sdk.Context,
 	portID, channelID string,
 	proposedUpgrade types.Upgrade,
+	flushStatus types.FlushStatus,
```
Comment (colin-axner) [inline modules/core/04-channel/keeper/upgrade.go:90]: What about writing the channel flush status as FLUSHING, and then in a followup pr we add a function call to check if flushing is completed?

**Review by colin-axner — APPROVED** [review]
LGTM, I left a suggestion for a slight modification. We can make the change later though if desired

File: modules/core/04-channel/keeper/upgrade.go (hunk 1 above)
Comment (charleenfei) [reply modules/core/04-channel/keeper/upgrade.go:90]: it's an interesting idea, i guess it depends on when we want to call the  `pendingInFlightPackets` function. i think we should discuss when we get to the `msg_server` implementation, we can modify it then if need be.

File: modules/core/04-channel/keeper/upgrade.go [hunk 2]
```diff
@@ -81,15 +81,30 @@ func (k Keeper) ChanUpgradeTry(
...
-	// TODO
-	// grab channel inside this function to get most current channel status
+	defer telemetry.IncrCounter(1, "ibc", "channel", "upgrade-try")
+
+	currentChannel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing channel when updating channel state in successful ChanUpgradeTry step, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	previousState := currentChannel.State
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade.go:99]: if we will be going ahead with https://github.com/cosmos/ibc/pull/978 then we can remove this var and hardcode to `INITUPGRADE` in the log statement below.

we can handle this later

**Review by damiannolan — APPROVED** [review]
Awesome, lgtm!

File: modules/core/04-channel/keeper/upgrade.go (hunk 1 above)
Comment (damiannolan) [reply modules/core/04-channel/keeper/upgrade.go:90]: Let's discuss tomorrow

**Pull request merged**